processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregatorFactory.java

```diff
@@ -94,11 +103,102 @@ public static Object estimateCardinality(Object object)
 
   private static final byte CACHE_TYPE_ID = (byte) 0x8;
   private static final byte CACHE_KEY_SEPARATOR = (byte) 0xFF;
+  private static final CardinalityAggregatorTypeHelperFactory TYPE_HELPER_FACTORY = new CardinalityAggregatorTypeHelperFactory();
+
 
   private final String name;
   private final List<DimensionSpec> fields;
   private final boolean byRow;
 
+  private static class CardinalityAggregatorTypeHelperFactory implements QueryTypeHelperFactory<CardinalityAggregatorTypeHelper>
+  {
+    @Override
+    public CardinalityAggregatorTypeHelper makeQueryTypeHelper(
+        String dimName, ColumnCapabilities capabilities
+    )
+    {
+      ValueType type = capabilities.getType();
+      switch(type) {
+        case STRING:
+          return new StringCardinalityAggregatorTypeHelper();
+        default:
+          return null;
```

Comment:
How valid is that this method returns null? Maybe add `@Nullable` to the interface method and document this?